openage/convert/dataformat/format_metadata.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2019-2019 the openage authors. See copying.md for legal info.
+
+
+class DataDefinition:
+    """
+    base class for sprite and terrain metadata collection and formatting
+    """
+    version = None
+
+    def __init__(self, name):
+        self.name = name
+        self.image_files = []
+        self.layers = []
+        self.frames = {}
+
+    def add_image(self, tup):
+        self.image_files.append(tup)
+
+    def add_layer(self, tup):
+        self.layers.append(tup)
+
+    def add_angle(self, degree, mirror_from=None):
+        if mirror_from is not None:
+            self.frames[degree] = mirror_from
+        else:
+            self.frames[degree] = []
+
+    def add_frame(self, angle, tup):
+        self.frames[angle].append(tup)
+
+    def dump(self):
+        raise NotImplementedError
+
+
+class SpriteDefinition(DataDefinition):
+    """
+    collects metadata for sprite definition
+    """
+    version = 0
+
+    def dump(self):
+        out = []
+        out.append('# openage sprite definition for {}\n'
+                   'version {}\n'.format(self.name, SpriteDefinition.version))
+        for image_id, file in self.image_files:
+            out.append('imagefile {} {}\n'.format(image_id, file))
+        for params in self.layers:
+            out.append('layer {} mode={} position={} time_per_frame={} replay_delay={}\n'.format(*params))
+        for degree, params in self.frames.items():
+            if isinstance(params, str):
+                out.append('angle {} mirror_from={}'.format(degree, params))
+            else:
+                out.append('angle {}\n'.format(degree))
+                for frame in params:
+                    out.append('frame {} {} {} {} {} {} {} {}\n'.format(*frame))
+        return ''.join(out)
+
+    def __str__(self):
+        return 'sprite metadata for: {}'.format(self.name)
+
+
+class TerrainDefinition(DataDefinition):
+    """
+    collects metadata for terrain definition
+    """
+    version = 0
+
+    def __init__(self, name):
+        super(TerrainDefinition, self).__init__(name)
```

Comment:
you do not need to pass `TerrainDefinition` and `self` as parameters here I think.